### ISSUE TITLE
line labels for 1D plotting

### ISSUE THREAD

**Issue by lamorton** [pydata/xarray#4255]
**Is your feature request related to a problem? Please describe.**
I find myself wanting to compare multiple 1D line plots.
```
d1 = xr.DataArray(np.ones((3,100)),dims=('channel','time'),coords={'channel':np.array(['a','b','c'])})
d1.sel(channel='a').plot()
d1.sel(channel='b').plot()
d1.sel(channel='c').plot()
```
The auto-magic labeling that happens when `_labels=True` (the default) is to place the title as `channel='a'` using the function `_title_for_slice()`.  This works fine when I only want to plot one line, but if I plot sequential lines, then the title gets over-written. So, I end up doing:

```
d1.sel(channel='a').plot(label='channel=a')
d1.sel(channel='b').plot(label='channel=c')
d1.sel(channel='c').plot(label='channel=b')
```
which is more boiler-plate than I'd like, and prone to errors (kudos if you noticed the one I made intentionally there).

**Describe the solution you'd like**
Basically, I'd just like to have these two lines added to the body of `xarray.plot.line` right around line 101:

```
    if _labels and not "label" in kwargs:
        kwargs["label"] = darray._title_for_slice()
```

This would automatically add the title as a label.  This would not change the way the plot looks in the original case, and if the caller of `plot` supplies a label, that will over-ride this default.

**Describe alternatives you've considered**
Another alternative is to make a switch in `plot` that would allow 2D arrays to be plotted as a series of 1D line plots like this, rather than as 2D heatmaps. It would be necessary to supply an indication of which of the two axes should be used as the x-axis of the plot, and which one would supply the line labels.

**Comment by dcherian** [comment]
See the `hue` kwarg: https://xarray.pydata.org/en/stable/plotting.html#multiple-lines-showing-variation-along-a-dimension

**Comment by lamorton** [comment]
Thanks @dcherian, that's what I'm looking for.
